### PR TITLE
Add Azure Function system module

### PR DESCRIPTION
**Pull request by WaelAbuSeada**

### CODE REVIEW

File: Modules/System/Azure Function/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            // Display the response
+            Message(ResponseTxt);
+        end
+        else
+            // Display the error message
+            Message(AzureFunctionResponse.GetError());
+    end;
+```
+
+#### Examle 2: send a GET request
```
Comment (brentholtorf) [inline Modules/System/Azure Function/README.md:44]: #### Example: Send a GET request

File: Modules/System/Azure Function/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+This module provides functionality for connecting to [Azure Functions](https://docs.microsoft.com/en-us/azure/azure-functions/).
+
+> This module does not store credentials for Azure Functions.  
+> Use caution when you store and pass credentials. 
+
+#### Examle: send a POST request
```
Comment (brentholtorf) [inline Modules/System/Azure Function/README.md:6]: #### Example: Send a POST request

File: Modules/System/Azure Function/app.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{
+    "id":  "8556c7d2-a594-4682-bd90-b10bfb176587",
+    "name":  "Connect to Azure Functions",
+    "publisher":  "Microsoft",
+    "brief":  "Simplify using Azure Functions",
+    "description":  "Support for writing AL code and connect to Azure Functions",
```
Comment (brentholtorf) [inline Modules/System/Azure Function/app.json:6]: "description":  "Support for writing AL code and connecting to Azure Functions",

File: Modules/System/Azure Function/AzureFunctionsImpl.Codeunit.al [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        QueryDict: Dictionary of [Text, Text];
+    begin
+        exit(Send(AzureFunctionAuthentication, Enum::"Http Request Type"::POST, QueryDict, Body, ContentTypeHeader));
+    end;
+
+    [NonDebuggable]
+    procedure Send(AzureFunctionAuthentication: Interface "Azure Functions Authentication"; RequestType: enum "Http Request Type"; QueryDict: Dictionary of [Text, Text]; Body: Text; ContentTypeHeader: text): Codeunit "Azure Functions Response"
+    var
+        Uri: Codeunit Uri;
+        UriBuider: Codeunit "Uri Builder";
```
Comment (pri-kise) [inline Modules/System/Azure Function/AzureFunctionsImpl.Codeunit.al:36]: There's a small typo. Should be `UriBuilder`

Comment (christianbraeunlich) [reply Modules/System/Azure Function/AzureFunctionsImpl.Codeunit.al:36]: If you haven't heard of it, https://marketplace.visualstudio.com/items?itemName=rasmus.al-var-helper lets you insert the variable name automatically - hope that helps, I learned about it last friday 📑 

As we are speaking about typos: there is also a missing s in the first parameter. Should be `AzureFunctionsAuthentication`

File: Modules/System/Azure Function/AzureFunctions.Codeunit.al [hunk 5]
```diff
@@ -0,0 +1,52 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 7804 "Azure Functions"
```
Comment (mazhelez) [inline Modules/System/Azure Function/AzureFunctions.Codeunit.al:6]: Some doc is missing here ;)

File: Modules/System/Azure Function/AzureFunctionsAuthentication.Codeunit.al [hunk 6]
```diff
@@ -0,0 +1,45 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 7800 "Azure Functions Authentication"
```
Comment (mazhelez) [inline Modules/System/Azure Function/AzureFunctionsAuthentication.Codeunit.al:6]: This one isn't documented as well 😮

File: Modules/System/Azure Function/AzureFunctionsAuthentication.interface.al [hunk 7]
```diff
@@ -0,0 +1,14 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+interface "Azure Functions Authentication"
```
Comment (mazhelez) [inline Modules/System/Azure Function/AzureFunctionsAuthentication.interface.al:6]: Add a few doc words for the interface as well

File: Modules/System/Azure Function/AzureFunctionsCodeAuth.Codeunit.al [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    begin
+        AuthenticationCodeGlobal := AuthenticationCode;
+        EndpointGlobal := Endpoint;
+    end;
+
+    [NonDebuggable]
+    procedure Authenticate(var RequestMessage: HttpRequestMessage): Boolean
+    var
+        Uri: Codeunit Uri;
+        UriBuider: Codeunit "Uri Builder";
```
Comment (tino-teuber) [inline Modules/System/Azure Function/AzureFunctionsCodeAuth.Codeunit.al:25]: `UriBuider` -> `UriBuilder`

File: Modules/System/Azure Function/README.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
+This module provides functionality for connecting to [Azure Functions](https://docs.microsoft.com/en-us/azure/azure-functions/).
+
+> This module does not store credentials for Azure Functions.  
+> Use caution when you store and pass credentials. 
+
```
Comment (brentholtorf) [inline Modules/System/Azure Function/README.md:5]: **NOTE:**
Azure Function Integration telemetry gathers data about the success or failure of calls from Business Central to Azure Functions. For more information, see [Analyzing Azure Function Integration Telemetry](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/administration/telemetry-azure-function-integration-trace?branch=kepontop-azure-function-telemetry).

**Review by JesperSchulz — APPROVED** [review]

**Pull request merged**